File: translations/codenet/deepseek_commented/C++2C/granite/Code/s189738089.c

#include <stdio.h>
#include <stdlib.h>
#define rep(i, k, n) for (int i = k; i < n; ++i)

int main(){
  // Initialization of the dp array
  int dp[101000] = {};
  dp[0] = 0;
  // input
  int N, K;
  scanf("%d %d", &N, &K);
  int h[N];
  rep(i, 0, N) scanf("%d", &h[i]);

  // Dynamic Programming Loop
  rep(i, 0, N) {
    rep(j, 1, K + 1) {
      dp[i + j] = dp[i + j] < dp[i] + abs(h[i] - h[i + j])? dp[i + j] : dp[i] + abs(h[i] - h[i + j]);
    }
  }

  // Output
  printf("%d\n", dp[N - 1]);
  return 0;
}

File: translations/codenet/deepseek_commented/C++2J/gpt-4o/Code/s189738089.java
import java.util.Scanner;

public class Main {
    static final int INF = 1 << 30;

    public static void main(String[] args) {
        // Initialization of the dp array
        int[] dp = new int[101000];
        for (int i = 0; i < dp.length; i++) {
            dp[i] = INF;
        }
        dp[0] = 0;

        // Input
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int K = scanner.nextInt();
        int[] h = new int[N];
        for (int i = 0; i < N; i++) {
            h[i] = scanner.nextInt();
        }

        // Dynamic Programming Loop
        for (int i = 0; i < N; i++) {
            for (int j = 1; j <= K; j++) {
                if (i + j < N) {
                    dp[i + j] = Math.min(dp[i + j], dp[i] + Math.abs(h[i] - h[i + j]));
                }
            }
        }

        // Output
        System.out.println(dp[N - 1]);
        scanner.close();
    }
}
// <END-OF-CODE>
